modules/sdk-core/src/bitgo/wallet/wallets.ts

```diff
@@ -554,9 +627,28 @@ export class Wallets implements IWallets {
     common.validateParams(params, ['walletShareId'], ['overrideEncryptedPrv', 'userPassword', 'newWalletPassphrase']);
 
     let encryptedPrv = params.overrideEncryptedPrv;
-
     const walletShare = (await this.getShare({ walletShareId: params.walletShareId })) as any;
-
+    if (walletShare.keychainOverrideRequired && walletShare.permissions.indexOf('admin') !== -1) {
+      if (_.isUndefined(params.userPassword)) {
+        throw new Error('userPassword param must be provided to decrypt shared key');
+      }
+      const walletKeychain = await this.createKeychain(params.userPassword);
+      const response = await this.updateShare({
+        walletShareId: params.walletShareId,
+        state: 'accepted',
+        keyId: walletKeychain.id,
+      });
+      // If the wallet share was accepted successfully (changed=true), reshare the wallet with the spenders
+      if (response.changed && response.state === 'accepted') {
+        try {
+          await this.reshareOfcAccountWithSpenders(walletShare.wallet, params.userPassword);
+        } catch (e) {
+          // TODO: gracefully handle this error
+          console.error(e);
```

Comment:
Address these TODO's

Reply:
Do you suggest adding a ticket or providing a thorough resolution?